### PR TITLE
More useful error output for type mismatches

### PR DESCRIPTION
**Pull request by philsturgeon**
Due to limitations with how should.js works, the type mismatches are currently giving rather confusing error messages. 

```
#/paths/~1/get/parameters/0/schema
expected true to be an array
    expected true to have [[Class]] Array
```

That is no good, as it doesn't mention _which_ property is true instead of being an array.

So! Instead lets make our own messages. Now it says:

```
#/paths/~1/get/parameters/0/schema
required should be an array
```

Simple, to the point, and when combined with #54 this'll be all the information anyone could need.

### CODE REVIEW

File: lib/validator.js [hunk 1]
```diff
@@ -158,58 +158,47 @@ function checkSubSchema(schema, parent, state) {
         if (typeof schema.additionalItems === 'boolean') {
         }
         else if (typeof schema.additionalItems === 'object') {
-            schema.additionalItems.should.not.be.an.Array();
+            schema.additionalItems.should.not.be.instanceOf(Array, 'additionalItems should not be an array, it shoud be a boolean or schema');
```
Comment (MikeRalphson) [inline lib/validator.js]: typo: shoud for should

File: lib/validator.js [hunk 2]
```diff
@@ -158,58 +158,47 @@ function checkSubSchema(schema, parent, state) {
...
         }
         else if (typeof schema.additionalProperties === 'object') {
-            schema.additionalProperties.should.not.be.an.Array();
+            schema.additionalProperties.should.not.be.instanceOf(Array, 'additionalProperties should not be an array, it should be a boolean or schema');
         }
         else should.fail(false,true,'additionalProperties must be a boolean or schema');
     }
     if (schema.maxItems) {
-        schema.maxItems.should.be.a.Number();
+        schema.maxItems.should.be.type(Number, 'maxItems should be a number');
```
Comment (MikeRalphson) [inline lib/validator.js]: Is this correct? Looking at https://shouldjs.github.io/#assertion-type shouldn't the type (1st parameter) be a string?

Comment (MikeRalphson) [reply lib/validator.js]: Several occurrences of the last issue.

**Comment by philsturgeon** [comment]
I'm going to avoid messing with the validator now that [oai-kit](https://raw.githubusercontent.com/Mermade/oas-kit) is a thing.

**Pull request closed without merging**